lib/compilers/dosbox-compiler.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2022, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import fs from 'fs-extra';
+
+import {BaseCompiler} from '../base-compiler';
+import * as exec from '../exec';
+import {logger} from '../logger';
+
+export class DosboxCompiler extends BaseCompiler {
+    private readonly dosbox: string;
+    private readonly root: string;
+
+    constructor(compilerInfo, env) {
+        super(compilerInfo, env);
+
+        this.dosbox = this.compilerProps(`compiler.${this.compiler.id}.dosbox`);
+        this.root = this.compilerProps(`compiler.${this.compiler.id}.root`);
+    }
+
+    protected override async writeMultipleFiles(files: any[], dirPath: string): Promise<any[]> {
+        const filesToWrite: any[] = [];
+
+        for (const file of files) {
+            if (!file.filename) throw new Error('One of more files do not have a filename');
+
+            const fullpath = this.getExtraFilepath(dirPath, file.filename);
+            const contents = file.contents.replaceAll(/\n/g, '\r\n');
```

Comment:
Oh boy of course! Wow :) And of course there's not a way to specify line-endings in `node` more generally.